### ISSUE TITLE
mlflow cli overrides experiment-id 

### ISSUE THREAD

**Issue by kyrre** [mlflow/mlflow#355]
The mlflow cli overrides whatever experiment-id you pass to the mlflow.start_run method. 

The minimal example below demonstrates the problem. In practice this means that it is not possible to let the user specify experiment-name and then set the associated experiment_id.

mlflow==0.5

main.py:
```
import click
import mlflow


@click.command()
@click.option("--experiment-id", required=True, type=int)
def main(experiment_id):
    with mlflow.start_run(experiment_id=experiment_id):
        pass


if __name__ == '__main__':
    main()

```

MLproject
```
name: dga

conda_env: model.yaml

entry_points:
  main:
    parameters:
      experiment-id: int

    command: "python main.py --experiment-id={experiment-id}"
```

model.yaml:
```
name: error
channels:
  - defaults
dependencies:
  - python=3.6
  - click
  - pip:
    - mlflow
```

**Comment by mparkhe** [comment]
Hi @kyrre : The interfaces are written with `experiment_id` and `run_uuid` ... etc in mind. In future we might look at additional arguments to support names within APIs.

There is a Pull Request in works [#122](https://github.com/mlflow/mlflow/pull/122) that is addressing a portion of this query to query experiment by name. Once this is merged in, could you use that in your wrapper method to pass an experiment name, look up the experiment ID and use that to start a run?

**Comment by kyrre** [comment]
Let me rephrase the problem as I see it. When you explicitly specify the experiment_id argument for the start_run method and then execute the project using the "mlflow run" command:

>  with mlflow.start_run(experiment_id=2):

then mlflow will end up using the default experiment_id (0). 

Is this the intended behaviour?
